### PR TITLE
YUNIKORN-489: remove unused message definitions

### PR DESCRIPTION
**Pull request by manirajv06**
Cleaning up unused message definitions.

### CODE REVIEW

File: scheduler-interface-spec.md [hunk 1]
```diff
@@ -910,27 +775,3 @@ message EventRecord {
...
-outstanding resource requests are there in the cluster, due to the shortage of node resources.
-This has excluded requests could not be satisfied because of other reasons, such as hitting queue
-resource limit, user/app limit, desire on certain placement constraints, etc.
-
-```protobuf
-// auto scaling metrics at a certain point of time
-message AutoScalingMetrics {
-  // a list of outstanding requests that desires for additional resources
-  repeated OutstandingResourceRequest outstandingRequests = 1;
-}
```
Comment (kingamarton) [inline scheduler-interface-spec.md:926]: I am not sure about this part. @yangwwei I see that you merged this change a few months ago from the autoscaling branch. Is this part of the autoscaling feature started a while ago, but not finished? If yes, I think we can remove it and we will add it back when we will really need it.

**Review by kingamarton — CHANGES_REQUESTED** [review]
Please check also the documentation, in case if there is some reference to the deleted messages, to make sure that there will be deleted from the documentation as well.

File: scheduler-interface-spec.md (hunk 1 above)
Comment (yangwwei) [reply scheduler-interface-spec.md:926]: hi @kingamarton  I think we can remove it as it is currently not used by any service.
We can always track back it won't be a problem when we want to resume the work on this in the future.

File: scheduler-interface-spec.md [hunk 2]
```diff
@@ -404,16 +404,10 @@ message AllocationAsk {
   int32 maxAllocations = 5;
   // Priority of ask
   Priority priority = 6;
-  // Execution timeout: How long this allocation will be terminated (by scheduler)
-  // once allocated by scheduler, 0 or negative value means never expire.
-  int64 executionTimeoutMilliSeconds = 7;
```
[the remark this answers is not in the document]
Comment (manirajv06) [reply scheduler-interface-spec.md]: Taken care

**Review by wilfred-s — APPROVED** [review]
LGTM
if there are no further comments then I will commit this in the next day or two.

**Review by kingamarton — APPROVED** [review]

**Pull request closed without merging**